File: doc/tutoNLP/memory_access/l4_field_write.ml
open Optitrust
open Prelude

(* Difficulty: Level 4
   Request: target the write to field x
   Expected target: [cFieldWrite ~field:"x" ()] *)

let _ = Run.script_cpp (fun () ->
  !! Show.target [cFieldWrite ~field:"x" ()];
)

File: doc/tutoNLP/memory_access/l4_field_write.cpp
// Difficulty: Level 4
// Request: target the write to field x

typedef struct {
  float x;
  float y;
} point;

void move(point* p) {
  p->x = p->x + 1.0f;
  p->y = p->y + 1.0f;
}
